source/agora/consensus/state/Ledger.d

```diff
@@ -383,6 +383,53 @@ public class Ledger
 
         if (header.height == this.last_block.header.height)
             this.last_block = this.storage.readLastBlock();
+
+        // reset the cached coinbase tx if next block is coinbase block
+        // and signature is for a block in that payout period
+        const next = this.height() + 1;
+        if (this.isCoinbaseBlock(next) && this.isRelatedRewardBlock(header.height, next))
```

Comment:
Two things.
- Isn't it better to call `isCoinbaseBlock` in `isRelatedRewardBlock`?
- Is there any reason to introduce the new variable, `next` instead of using `this.height()` in place?

Reply:
I think keeping the call to `isCoinbaseBlock` here is better to document what we are checking. 
Otherwise the name for the other function needs to be quite verbose, e.g. `nextBlockIsCoinbaseAndHeaderBlockisRelatedRewardBlock`.